exporters/trace/gcp_exporter/internal/recordable_test.cc

```diff
@@ -1,12 +1,82 @@
 #include "exporters/trace/gcp_exporter/recordable.h"
 #include <gtest/gtest.h>
 
+
 OPENTELEMETRY_BEGIN_NAMESPACE
 namespace exporter
 {
 namespace gcp
 {
 
+using cloudtrace_v2_span = google::devtools::cloudtrace::v2::Span;
+constexpr char kLongRegularString[] = "Some long string to be truncated";
+constexpr char kLongUnicodeString[] = "些长字符串被截断 Некоторая длинная строка, подлежащая усечению";
+
+class RecordableTestPeer: public testing::Test {
+public:
+    void SetTruncatableString(cloudtrace_v2_span& span,
```

Comment:
Here we are testing private implementation, instead of testing public methods.
It would be better to run tests against SetName and SetAttribute instead. This will also remove the need to run make test a friend of a class.

Reply:
I think we also need to set the `limit` parameter, which in the tests is `max_size_bytes`. 
`SetName` and `SetAttribute` do not have `limit` as a parameter.

Reply:
I think this is part of the problem with testing internal methods. 
If we were testing public interface, we need ensure that Name that exceeds kDisplayNameStringLen gets truncated by utf8 symbol, and all attributes are truncated at kAttributeStringLen or before it by utf8 syymbol.
If we test for that, there is no need to set limit in tests.

Private implementation can potentially change in future, so testing against it is fragile, we only need to test the API via public methods here. Thsi would also allow hiding implementation in cc file entirely.

Reply:
I see.
Sure!